test.js

```diff
@@ -1581,6 +1581,285 @@ test('errors', function (t) {
     })
 })
 
+test('token in cookie, with fastify-cookie parsing', function (t) {
+  t.plan(6)
+
+  const fastify = Fastify()
+  fastify.register(jwt, { secret: 'test', cookie: { cookieName: 'jwt' } })
+  fastify.register(require('fastify-cookie'))
+
+  fastify.post('/sign', function (request, reply) {
+    reply.jwtSign(request.body)
+      .then(function (token) {
+        return reply.send({ token })
+      })
+  })
+
+  fastify.get('/verify', function (request, reply) {
+    request.jwtVerify()
+      .then(function (decodedToken) {
+        return reply.send(decodedToken)
+      })
+      .catch(function (error) {
+        return reply.send(error)
+      })
+  })
+
+  fastify
+    .ready()
+    .then(function () {
+      t.test('token present in cookie', function (t) {
+        t.plan(2)
+        fastify.inject({
+          method: 'post',
+          url: '/sign',
+          payload: { foo: 'bar' }
+        }).then(function (signResponse) {
+          const token = JSON.parse(signResponse.payload).token
+          t.ok(token)
+
+          fastify.inject({
+            method: 'get',
+            url: '/verify',
+            cookies: {
+              jwt: token
+            }
+          }).then(function (verifyResponse) {
+            const decodedToken = JSON.parse(verifyResponse.payload)
+            t.is(decodedToken.foo, 'bar')
+          }).catch(function (error) {
+            t.fail(error)
+          })
+        }).catch(function (error) {
+          t.fail(error)
+        })
```

Comment:
remove the catch and return the promise, so tap would do this for us automatically